packages/neo-one-editor/src/manager/PackageManager.ts

```diff
@@ -0,0 +1,146 @@
+import { FileSystem } from '@neo-one/local-browser';
+import fetch from 'cross-fetch';
+import * as path from 'path';
+import { Observable, Subscription } from 'rxjs';
+import { distinctUntilChanged, map, switchMap } from 'rxjs/operators';
+import { Dependencies, DependencyInfo, PackageJSON, ResolvedDependencies, Resolver } from './Resolver';
+
+const PACKAGE_JSON = 'package.json';
+
+interface FetchPackageInfo {
+  readonly name: string;
+  readonly version: string;
+  readonly file?: string;
+}
+
+interface NMPackageInfo {
+  readonly depInfo: DependencyInfo;
+  readonly pkgPath: string;
+}
+
+interface Module {
+  readonly name: string;
+  readonly fileName: string;
+  readonly version: string;
+  readonly dependency: string;
+  readonly packageJSON: string;
+  readonly types?: string;
+  readonly typeFile: string;
+}
+
+export class PackageManager {
+  public readonly dependencies$: Observable<ResolvedDependencies>;
+  public readonly packages$: Observable<{ readonly [name: string]: Module }>;
+  private readonly fs: FileSystem;
+  private readonly subscription: Subscription;
+
+  public constructor({
+    fs,
+    packageJSON$,
+  }: {
+    readonly packageJSON$: Observable<PackageJSON>;
+    readonly fs: FileSystem;
+  }) {
+    this.fs = fs;
+    this.dependencies$ = packageJSON$.pipe(
+      distinctUntilChanged(),
+      switchMap(async (packageJSON) =>
+        this.resolveDependencies(packageJSON.dependencies === undefined ? {} : packageJSON.dependencies),
+      ),
+    );
+    this.packages$ = this.dependencies$.pipe(switchMap(async (deps) => this.fetchPackages(deps, 'node_modules')));
```

Comment:
Make `pkgPath` be `/node_modules` so it doesn't resolve in the tests to a local directory (and instead, resolves to `/node_modules/....`)